### PR TITLE
Fix group add/remove membership bug

### LINKED ISSUE #8841: Remove of groups does not work properly
If i remove a user from a group, the user is not removed at all nodes in the cluster.
i remove the user with the command "mc admin group remove" from a windows 10 client. the server runs under ubuntu 18.04

command to remove
mc admin group remove minio mygroup myuser

command to check
mc admin user info minio myuser

Here an example out put:

```
D:\MinIo>mc admin user info exanic developer
AccessKey: developer
Status: enabled
PolicyName:
MemberOf: heidak-dev-write,newhome-build-write,newhome-dev-write,newhome-integration-write

D:\MinIo>mc admin user info exanic developer
AccessKey: developer
Status: enabled
PolicyName:
MemberOf: heidak-dev-write,newhome-build-write,newhome-dev-write

D:\MinIo>mc admin user info exanic developer
AccessKey: developer
Status: enabled
PolicyName:
MemberOf: heidak-dev-write,newhome-build-write,newhome-dev-write,newhome-integration-write

D:\MinIo>mc admin user info exanic developer
AccessKey: developer
Status: enabled
PolicyName:
MemberOf: heidak-dev-write,newhome-build-write,newhome-dev-write,newhome-integration-write

D:\MinIo>mc admin user info exanic developer
AccessKey: developer
Status: enabled
PolicyName:
MemberOf: heidak-dev-write,newhome-dev-write,newhome-integration-write,newhome-staging-write

D:\MinIo>mc admin user info exanic developer
AccessKey: developer
Status: enabled
PolicyName:
MemberOf: heidak-dev-write,newhome-build-write,newhome-dev-write,newhome-integration-write,newhome-staging-write
```
The commands were carried out one after the other

EDIT:  The Cluster is behind a loadbalancer ... So you can see many different nodes

## Expected Behavior
the user should removed from the group at all nodes inside the cluster

## Current Behavior
it looks like, that the user is only remove at the node it the command executed. In the logs at the server i cant found any exception

## Steps to Reproduce (for bugs)
1. Create a user and a group
2. Add the user to the group
3. Remove the user from the group
4. Check if the user is removed

## Your Environment

* Version used (`minio version`): 2020-01-16T22:40:29Z
* Environment name and version (e.g. nginx 1.9.1): 
* Server type and version: ubuntu 18.04 lts
* Operating System and version (`uname -a`): Linux vstorage01 4.15.0-62-generic #69-Ubuntu SMP Wed Sep 4 20:55:53 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

### PR DESCRIPTION
**Pull request by donatello**
## Description

Fix https://github.com/minio/minio/issues/8841

## How to test this PR?

Same steps as in the linked issue.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed
- [ ] Functional tests needed (If yes, add [mint](https://github.com/minio/mint) PR # here: )

### CODE REVIEW

**Comment by donatello** [comment]
The changes to go.mod and go.sum appeared automatically on `make`.

File: cmd/iam.go [hunk 1]
```diff
@@ -1449,9 +1449,6 @@ func (sys *IAMSys) updateGroupMembershipsMap(group string, gi *GroupInfo) {
 // removeGroupFromMembershipsMap - removes the group from every member
 // in the cache. IMPORTANT: Assumes sys.Lock() is held by caller.
 func (sys *IAMSys) removeGroupFromMembershipsMap(group string) {
-	if _, ok := sys.iamUserGroupMemberships[group]; !ok {
-		return
-	}
```
Comment (donatello) [inline cmd/iam.go:1454]: The 3 lines above are the cause of the bug. `group` is not a valid key type for this map.

File: go.mod [hunk 2]
```diff
@@ -111,6 +111,7 @@ require (
 	golang.org/x/oauth2 v0.0.0-20190402181905-9f3314589c9a // indirect
 	golang.org/x/sys v0.0.0-20190922100055-0a153f010e69
 	golang.org/x/text v0.3.2 // indirect
+	golang.org/x/tools/gopls v0.2.2 // indirect
```
Comment (harshavardhana) [inline go.mod]: This doesn't look like we directly depend `// indirects` can be removed.

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#95841bc5-5f7a-42ec-901b-12e2f342408c) |
<a name="95841bc5-5f7a-42ec-901b-12e2f342408c"></a>
### 8877-9ceaa61/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.61:30238
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 39b5b3c698a7:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... done in 2 minutes and 52 seconds
(2/13) Running aws-sdk-java tests ... done in 2 seconds
(3/13) Running aws-sdk-ruby tests ... done in 8 seconds
(4/13) Running mc tests ... done in 56 seconds
(5/13) Running minio-go tests ... FAILED in 3 minutes and 53 seconds
{
  "args": {},
  "duration": 106271,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 4 out of 13 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Review by harshavardhana — APPROVED** [review]

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-xl.sh | :x: [more...](#7cb87143-081d-4d10-8eef-8b9aebd6b08c) |
| mint-large-bucket.sh | :x: [more...](#e886e5e2-0f12-436e-ad86-81c5bc4b18a9) |
<a name="7cb87143-081d-4d10-8eef-8b9aebd6b08c"></a>
### 8877-e9f800d/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.58:31389
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 1c7f5ccbd033:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... FAILED in 12 seconds
{
  "name": "awscli",
  "duration": 9645,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "An error occurred (XMinioServerNotInitialized) when calling the CreateBucket operation (reached max retries: 4): Server not initialized, please try again."
}

Executed 0 out of 13 tests successfully.
```
<a name="e886e5e2-0f12-436e-ad86-81c5bc4b18a9"></a>
### 8877-e9f800d/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.61:32122
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 712c07452fd5:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... FAILED in 11 seconds
{
  "name": "awscli",
  "duration": 8924,
  "function": "delete_bucket\n",
  "status": "FAIL",
  "error": "fatal error: An error occurred (XMinioServerNotInitialized) when calling the ListObjectsV2 operation (reached max retries: 4): Server not initialized, please try again.remove_bucket failed: Unable to delete all objects in the bucket, bucket will not be deleted."
}

Executed 0 out of 13 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Review by poornas — APPROVED** [review]
LGTM

**Pull request merged**